api/data_test.go

```diff
@@ -191,6 +191,27 @@ func TestCreateNewCacheTime(t *testing.T) {
 		})
 	})
 }
+
+func TestGetCacheTimeReturnsError400(t *testing.T) {
+	Convey("Given an API", t, func() {
+		ctx := context.Background()
+		dataStoreMock := &mock.DataStoreMock{}
+		dataStoreAPI := setupAPIWithStore(ctx, dataStoreMock)
+		Convey("When the id provided is not 32 characters, not hexidecimal and not lowercase", func() {
+			var idTotallyInvalid = "XXX"
+			request := httptest.NewRequest(http.MethodGet, baseURL+idTotallyInvalid, http.NoBody)
+			responseRecorder := httptest.NewRecorder()
+			dataStoreAPI.Router.ServeHTTP(responseRecorder, request)
+			Convey("Then a 400 is returned with all three ID validation errors raised", func() {
+				So(responseRecorder.Code, ShouldEqual, 400)
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "id should be 32 characters in length")
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "id is not a valid hexadecimal")
+				So(responseRecorder.Body.String(), ShouldContainSubstring, "id is not lowercase")
+			})
+		})
+	})
+}
```

Comment:
Would suggest adding a component test for this as well if Jose's other PR has been merged.